### PR TITLE
chore: introduce i18next

### PR DESCRIPTION
**Pull request by rot1024**
# Overview

## What I've done

- Introduce i18nnext
- Add a npm script: `yarn i18n` (so fast!!!)
- Rename `gen:18n` script to `i18n:legacy`
- Rename `gen:gql` script to `gql` (fewer type times)

## What I haven't done

react-intl is not uninstalled yet. The number of translations is so large that it would be difficult to migrate them all at once, so we would like to rewrite them using i18next when we see react-intl when we revamp a component.

## How I tested

Check Netlify preview to confrim the app continues to work

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+    "editor.formatOnSave": false, // conflicts with ESLint format
```
Comment (KaWaite) [inline .vscode/settings.json:2]: If other devs need to update/revert their IDE settings, please let us know on slack!

File: i18next-parser.config.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  locales: ["en", "ja"],
+  output: "src/translations/$LOCALE.yml",
```
Comment (KaWaite) [inline i18next-parser.config.js:3]: Is there any reason we will keep translations separated from i18n code? In CMS and Cloud we are storing everything together, so if there is a good reason to keep them separate I will update CMS and cloud to keep things streamlined. If not, please output to 
```suggestion
  output: "src/i18n/translations/$LOCALE.yml",
```

File: package.json [hunk 3]
```diff
@@ -24,9 +24,9 @@
         "storybook:screenshot": "storycap --serverCmd 'npx http-server ./storybook-static -p 6006' --serverTimeout 100000 http://localhost:6006 --max-old-space-size=2048",
         "cypress:open": "cypress open",
         "cypress:run": "cypress run",
-        "gen": "run-p gen:*",
-        "gen:gql": "graphql-codegen --config codegen.yml",
-        "gen:i18n": "extract-messages -d en -l=en,ja -o translations -f yaml 'src/**/*.ts*' --extractFromFormatMessageCall",
+        "gql": "graphql-codegen --config codegen.yml",
+        "i18n": "i18next",
+        "i18n:regacy": "extract-messages -d en -l=en,ja -o src/translations/legacy -f yaml 'src/**/*.ts*' --extractFromFormatMessageCall",
```
Comment (KaWaite) [inline package.json:29]:
```suggestion
        "i18n:legacy": "extract-messages -d en -l=en,ja -o src/translations/legacy -f yaml 'src/**/*.ts*' --extractFromFormatMessageCall",
```

File: src/i18n/publishedProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React, { PropsWithChildren } from "react";
+import { I18nextProvider } from "react-i18next";
+
+import i18n from "./i18n";
+import { PublishedProvider as LegacyProvider } from "./legacy";
+
+export default function PublishedProvider({ children }: PropsWithChildren<{}>) {
```
Comment (KaWaite) [inline src/i18n/publishedProvider.tsx:7]: Based on [this post](https://stackoverflow.com/a/71809927) by one of React's devs, he says PropsWithChildren doesn't need to be used, and is only useful for the `codemod`. 
Also, children typings are fixed in React 18, so we will need to start providing a prop type. Can you add that here? Type for children: `children?: React.ReactNode`

File: src/translations/en.yml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+Continue: ''
+Cancel: ''
```
Comment (KaWaite) [inline src/translations/en.yml:2]: Is this correct? I haven't had a chance to look into i18next enough yet, but I was hoping we could setup the generation to be better than how it is in cloud (manual) and cms (I think same as here). Benefits of this current way is the Japanese will be easier to translate (bc the english can be seen in ja.yml), so maybe this way is ok? Just `EnglishWord: ''` seems a little strange to me.

File: src/i18n/provider.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import { useLanguageQuery } from "@reearth/gql";
+
+import i18n from "./i18n";
+import { Provider as LegacyProvider } from "./legacy";
+
+export default function Provider({ children }: { children?: ReactNode }) {
+  const { isAuthenticated } = useAuth();
+  const { data } = useLanguageQuery({ skip: !isAuthenticated });
+  const locale = data?.me?.lang;
+  useEffect(() => {
```
Comment (KaWaite) [inline src/i18n/provider.tsx:14]:
```suggestion
  const locale = data?.me?.lang;
  
  useEffect(() => {
```

File: src/i18n/provider.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    i18n.changeLanguage(locale);
```
Comment (KaWaite) [inline src/i18n/provider.tsx:15]: `locale`
Screen Shot 2022-04-27 at 11 42 20
, if not chosen by the user yet, apparently can be `und` which is used for the `Auto` setting.

**Review by KaWaite — COMMENTED** [review]
Reviewed!

File: .vscode/settings.json (hunk 1 above)
Comment (rot1024) [reply .vscode/settings.json]: OK, let me discuss about it later. For now, that line has been reverted.

File: src/translations/en.yml (hunk 5 above)
Comment (rot1024) [reply src/translations/en.yml]: Setting useKeysAsDefaultValue to true will cause the key name to be inserted directly into the value.

File: src/i18n/provider.tsx (hunk 7 above)
Comment (rot1024) [reply src/i18n/provider.tsx]: I tried to set undefined as the language (i.e. automatic) when und. But it needs to be tested.

Comment (rot1024) [reply src/i18n/provider.tsx]: tested and it works

**Review by KaWaite — APPROVED** [review]
Looks good!

**Pull request merged**